src/service_cmd/runner/runner.go

```diff
@@ -75,6 +77,16 @@ func createLimiter(srv server.Server, s settings.Settings, localCache *freecache
 
 func (runner *Runner) Run() {
 	s := runner.settings
+	if s.TracingEnabled {
```

Comment:
I want to make sure I understand how this is implemented: Right now all of these traces are independent, right? Because we aren't passing any incoming span data, right? Are you still planning on changing the API to optionally pass on context?

Reply:
Right now all spans in one call to `ShouldRateLimit` gRPC function are associated in one trace using the context (e.g. gRPC client span + gRPC server span + redis pipeline execution span, you can refer to screenshot above). But in the real use case where it is the envoyproxy calling the ratelimit, the envoyproxy needs to instrument the gRPC call to ratelimit service in order to associate the trace.